### ISSUE TITLE
[Date field] Focus indicator stays on date field when user's focus is inside dialog

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3134]
## Current behavior 

Focus indicator is placed on whole `sl-date-field` when user enters editable spinbuttons inside it. When user moves focus to `sl-field-button` "Select date" focus indicator is both on the date field and button. If user opens dialog one focus indicator goes to the first selectable date but another stays on Date field. 
This behavior might be confusing to some sighted keyboard users.

https://github.com/user-attachments/assets/80bd0694-0e22-465a-b753-969f17d33024

## Suggested solution

There should always be only one focus indicator visible.

- When user is on day/month/year spinbuttons focus indicator should be on Date field.
- When user is on "Select date" button focus indicator should be on that button.
- When user is inside dialog focus indicator should be inside dialog.

### 👤 Your name

Kacper

### 🧢 Your Product/Team

SLDS
